common/random/shuffle.go

```diff
@@ -0,0 +1,28 @@
+package random
+
+import (
+	"math/rand"
+)
+
+func Shuffle[T any](slice []T) {
+	//nolint:gosec
+	r := rand.New(CryptoSource{}) //CryptoSource is a cryptographic source of randomness
+
+	r.Shuffle(len(slice), func(i, j int) {
+		slice[i], slice[j] = slice[j], slice[i]
+	})
+}
+
+// ShuffleMap returns a shuffled slice of keys to iterate by
+func ShuffleMap[K comparable, V any](m map[K]V) []K {
```

Comment:
This isn't deterministically random I guess?

Reply:
As far as it uses crypto/rand, it isn't.

Reply:
This looks like a good explanation: https://stackoverflow.com/a/55925880

Reply:
Yeap, although it only depends on non cryptographically strong hash and a seed. The PR introduces cryptographically strong solution.